### DISCUSSION TITLE
PCの個別環境によるログインの失敗について

### DISCUSSION THREAD

**Discussion by nogogo** [pixiv/vroid-sdk-developers#54] [Q&A] [answered]
お世話になっております。

アプリ内でVroidHubにログインする過程の、
①ブラウザでVroidHubログイン
→②ブラウザで認可コード表示
→③認可コードをアプリ側で入力する
→④アプリ内でVroidHubにログイン（連携）
の、④において、ファイアウォールが原因で、ログインができない、というエラーはあり得ますでしょうか。
また、その他個別PC環境のせいでアプリ内ログインができない、などの例はございますでしょうか。
現在、「認可コード表示・入力まではできるが、アプリ内でログインができない」との不具合報告が一件だけ来ており、その原因を調査している経緯であります。
尚、ログインできない不具合は今のところこの一件のみですので、固有のPC環境のせいなのかなと思っております。

ご回答いただけますと幸いです。

**Comment by nyamadan** [comment, marked as answer]
通信を行う都合ファイアウォールが連携に影響を与える可能性は考えられますが、特定の製品でそういったことが発生しているという情報は今のところ手元には来ておりません。

他にはログイン関連情報の保存に失敗すると同様の問題が発生する可能性がございます。
稀なケースだとは思いますがこういった事象が発生した場合にその旨をユーザーに通知する、または、保存失敗の影響を緩和する方法を検討中です。

**Reply by nogogo** [reply]
ご回答ありがとうございます。
>他にはログイン関連の保存に失敗すると同様の問題が発生する可能性がございます

こちらについて、現状アプリ側でできる対策はありますでしょうか。それともしょうがない事なのでしょうか。

**Reply by nyamadan** [reply]
> こちらについて、現状アプリ側でできる対策はありますでしょうか。それともしょうがない事なのでしょうか。

お待たせいたしました。

保存に問題があると仮定し、保存するタイミングで try - catch をすればよいと思うのですがアプリ側から対応できる箇所を見つけることができませんでした。

ライブラリ側での対応を進めますのでお待ちいただければと思います。
